includes/blocks/class-sensei-block-take-course.php

```diff
@@ -0,0 +1,170 @@
+		add_action( 'init', [ $this, 'register_block' ] );
+	}
+
+
+	/**
+	 * Register progress bar block.
+	 *
+	 * @access private
+	 */
+	public function register_block() {
+		register_block_type(
+			'sensei-lms/button-take-course',
+			[
+				'render_callback' => [ $this, 'render_take_course_block' ],
+			]
+		);
+	}
+
+	/**
+	 * Render the take course button. Wraps block HTML within a form.
+	 *
+	 * @param array  $attributes Block attributes.
+	 * @param string $content    Block HTML.
+	 *
+	 * @return string Form to start the course, with the block content as the submit button.
+	 */
+	public function render_take_course_block( $attributes, $content ): string {
+		global $post;
+		$course_id = $post->ID;
+
+		if ( Sensei_Course::can_current_user_manually_enrol( $course_id ) ) {
+			if ( ! Sensei_Course::is_prerequisite_complete( $course_id ) ) {
+				return $this->render_disabled_with_prerequisite( $course_id, $content );
+			}
+			return $this->render_with_start_course_form( $course_id, $content );
+		}
+
+		if ( ! is_user_logged_in() ) {
+			return $this->render_with_login( $content );
+		}
+
+		return '';
+
+	}
+
+	/**
+	 * Render block with start course action.
+	 *
+	 * @param int    $course_id
+	 * @param string $button Block HTML.
+	 *
+	 * @return string
+	 */
+	private function render_with_start_course_form( $course_id, $button ) {
+		$nonce = wp_nonce_field( 'woothemes_sensei_start_course_noonce', 'woothemes_sensei_start_course_noonce', false, false );
+		return ( '
+			<form method="POST" action="' . esc_url( get_permalink( $course_id ) ) . '">
+			<input type="hidden" name="course_start" value="1" />
+			' . $nonce . '
+			' . $button . '
+			</form>
+			' );
+	}
+
+	/**
+	 * Render block with link to login page, when the user is not logged in.
+	 *
+	 * @param string $content Block HTML.
+	 *
+	 * @return string
+	 */
+	private function render_with_login( $content ) {
+
+		/**
+		 * Filter to force Sensei to output the default WordPress user
+		 * registration link.
+		 *
+		 * @param bool $wp_register_link default false
+		 *
+		 * @since 1.9.0
+		 */
+		$wp_register_link = apply_filters( 'sensei_use_wp_register_link', false );
+
+		$settings = Sensei()->settings->get_settings();
+		if ( ! empty( $settings['my_course_page'] ) && ! $wp_register_link ) {
+			$my_courses_url = get_permalink( intval( $settings['my_course_page'] ) );
+			$target         = esc_url( $my_courses_url );
+		} else {
+			$target = wp_registration_url();
+		}
+
+		return ( '
+			<form method="GET" action="' . esc_url( $target ) . '">
+			' . $content . '
+			</form>
+			' );
+	}
+
+	/**
+	 * Render disabled state when a prerequisite course is required.
+	 *
+	 * @param int    $course_id
+	 * @param string $content Block HTML.
+	 *
+	 * @return string
+	 */
+	private function render_disabled_with_prerequisite( $course_id, $content ) {
+		$notice  = '<figcaption>' . $this->get_course_prerequisite_message( $course_id ) . '</figcaption>';
```

Comment:
Maybe `small` could be a little more semantic here. WDYT?

Reply:
Yeah this is a bit off, I wanted this to pick up some theme styles that mostly look right for the part. But maybe we can just style it with CSS.